transitclock/src/main/java/org/transitclock/utils/threading/NamedThread.java

```diff
@@ -141,6 +146,7 @@ public void run() {
 				System.exit(-1);
 			}
 		} finally {
+			l.unlock();
```

Comment:
ok, you really need to convince me this is necessary.  Locking an entire thread seems completely unreasonable.  I suspect you don't understand how this thread is used.

Reply:
I can remove this and retest.